### PR TITLE
Add changes to expose metric for installed knative version

### PR DESCRIPTION
**Pull request by savitaashture**
JIRA: https://jira.coreos.com/browse/SRVKS-187

Description:
This PR include changes to expose installed knative version information to metric and servicemonitor.

Screenshot from 2019-08-20 18-31-10

### CODE REVIEW

File: pkg/controller/knativeserving/common/extensions.go [hunk 1]
```diff
@@ -5,13 +5,14 @@ import (
...
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
 var log = logf.Log.WithName("common")
 
-type Platforms []func(client.Client, *runtime.Scheme, *mf.Manifest) (*Extension, error)
+type Platforms []func(client.Client, *runtime.Scheme, *mf.Manifest, *rest.Config) (*Extension, error)
```
Comment (markusthoemmes) [inline pkg/controller/knativeserving/common/extensions.go]: Is all the config passing still necessary?

Comment (savitaashture) [reply pkg/controller/knativeserving/common/extensions.go]: No its not required now :)

**Comment by savitaashture** [comment]
All Review comments addressed from this PR https://github.com/openshift-knative/knative-serving-operator/pull/49

File: pkg/controller/knativeserving/openshift/openshift.go [hunk 2]
```diff
@@ -32,6 +32,11 @@ const (
 
 	// The secret in which the tls certificate for the autoscaler will be written.
 	autoscalerTlsSecretName = "autoscaler-adapter-tls"
+	// knativeServingInstalledNamespace is the ns where knative serving operator has been installed
+	knativeServingInstalledNamespace = "NAMESPACE"
+	// service monitor created successfully when monitoringLabel added to namespace
+	monitoringLabel = "openshift.io/cluster-monitoring"
+	rolePath        = "deploy/resources/monitoring/role_service_monitor.yaml"
```
Comment (nak3) [inline pkg/controller/knativeserving/openshift/openshift.go]: Now, this rolePath should be moved into `installServiceMonitor()`.

Comment (savitaashture) [reply pkg/controller/knativeserving/openshift/openshift.go]: Done

File: pkg/controller/knativeserving/openshift/openshift.go [hunk 3]
```diff
@@ -92,46 +96,32 @@ func serviceMonitorExists(namespace string) (bool, error) {
 }
 
 func installServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	const (
+		path         = "deploy/resources/monitoring/service_monitor.yaml"
+		operatroPath = "deploy/resources/monitoring/operator_service_monitor.yaml"
```
Comment (markusthoemmes) [inline pkg/controller/knativeserving/openshift/openshift.go]:
```suggestion
		operatorPath = "deploy/resources/monitoring/operator_service_monitor.yaml"
```

Comment (savitaashture) [reply pkg/controller/knativeserving/openshift/openshift.go]: Done

**Pull request merged**